R/prepare-lpr.R

```diff
@@ -81,33 +81,86 @@ prepare_lpr2 <- function(lpr_adm, lpr_diag) {
     )
 }
 
-#' Join together the LPR3 (`diagnoser` and `kontakter`) registers.
+#' Prepare and join the two LPR3 registers to extract diabetes and pregnancy diagnoses.
 #'
-#' To prepare the LPR3 data for the inclusion process, this function also
-#' renames the `cpr` variable to `pnr` to match the other registers.
+#' The output is used as inputs to `include_diabetes_diagnoses()` and to
+#' `get_pregnancy_dates()` (see exclusion events).
+#'
+#' @param diagnoser The LPR3 register containing diabetes diagnoses.
+#' @param kontakter The LPR3 register containing hospital contacts/admissions.
 #'
-#' @param diagnoser The diagnosis register.
-#' @param kontakter The contacts register.
+#' @return The same type as the input data, default as a [tibble::tibble()],
+#'  with the following columns:
+#'
+#'  -   `pnr`: The personal identification variable.
+#'  -   `date`: The date of all the recorded diagnosis (renamed from
+#'      `d_inddto`).
+#'  -   `is_primary_dx`: Whether the diagnosis was a primary diagnosis.
+#'  -   `is_diabetes_code`: Whether the diagnosis was any type of diabetes.
+#'  -   `is_t1d_code`: Whether the diagnosis was T1D-specific.
+#'  -   `is_t2d_code`: Whether the diagnosis was T2D-specific.
+#'  -   `is_endocrinology_dept`: Whether the diagnosis was made by an
+#'      endocrinology medical department.
+#'  -   `is_medical_dept`: Whether the diagnosis was made by a
+#'      non-endocrinology medical department.
+#'  -   `is_pregnancy_code`: Whether the person has an event related to
+#'       pregnancy like giving birth or having a miscarriage at the given date.
```

Comment:
```suggestion
#' @inherit prepare_lpr2 return
```

So that there is one set of documentation to use and maintain.

Reply:
Ooo, nice!